### PR TITLE
chawan: 0-unstable-2024-03-01 -> 0-unstable-2024-07-14; set updateScript

### LINKED ISSUE #327404: Build failure: chawan
### Steps To Reproduce

1. build *chawan* (4f7221b0e20daf9282024d00fb8463f6973441b7)

### Build log

```
nim c --nimcache:".obj/release/cha" -d:libexecPath='${%CHA_BIN_DIR}/../libexec/chawan' \
        -d:release -d:strip -d:lto -o:"target/release/bin/cha" src/main.nim
Hint: used config file '/nix/store/k0apz12k1a4j1yp21qm8wn5amdn80p4a-x86_64-unknown-linux-gnu-nim-wrapper-2.0.8/etc/nim/nim.cfg' [Conf]
Hint: used config file '/nix/store/k0apz12k1a4j1yp21qm8wn5amdn80p4a-x86_64-unknown-linux-gnu-nim-wrapper-2.0.8/etc/nim/config.nims' [Conf]
Hint: used config file '/build/source/nim.cfg' [Conf]
............................................................................................................................
/build/source/src/bindings/quickjs.nim(64, 28) Error: object with '{.union.}' pragmas is not supported by VM
make: *** [Makefile:54: target/release/bin/cha] Error 1
lines 385-420/420 (END)
```

Related to https://github.com/nim-lang/Nim/pull/23362

### Notify maintainers

@jtbx

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

Set updateScript and run it.
Also, remove perl as it was used to generate the manpages via an older process - necessitating changing the patch as well.

Also fixes the build failure - fixes #327404 (highlighted also in https://github.com/NixOS/nixpkgs/pull/321092#issuecomment-2179961895)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ch/chawan/mancha-augment-path.diff [hunk 1]
```diff
@@ -2,14 +2,14 @@ Add the -m option to man's command line to augment the list of paths
 searched by man. The string "OUT" must be substituted with chawan's $out
 path after patching.
```
Comment (jtbx) [inline pkgs/by-name/ch/chawan/mancha-augment-path.diff:3]: You can remove this
```suggestion
searched by man.
```

**Review by jtbx — APPROVED** [review]
Works well, thank you!

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 327421` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>chawan</li>
  </ul>
</details>

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 327421` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>chawan</li>
  </ul>
</details>

**Pull request merged**